### PR TITLE
Add MainQueue and PreferenceQueue

### PR DESCRIPTION
**Pull request by jackedelic**

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -106,8 +111,12 @@ public void addDoctor(Doctor doctor) {
         indicateAddressBookChanged();
     }
     @Override
-    public void enqueue(Person target) {
-        patientQueue.add(target);
+    public void enqueue(Person patient) {
+        mainQueue.add(patient);
+    }
+    @Override
```
Comment (gingivitiss) [inline src/main/java/seedu/address/model/ModelManager.java]: Requires newline before override.

File: src/main/java/seedu/address/model/Model.java [hunk 2]
```diff
@@ -151,7 +151,7 @@
 
     /**
      * Updates the filter of the filtered appointment list to filter by the given {@code predicate}.
-     * @throws NullPointerException if {@code predicate} is null.
+     * @throwstat NullPointerException if {@code predicate} is null.
```
Comment (jjlee050) [inline src/main/java/seedu/address/model/Model.java]: How come got throwstat? Isnt it suppose to be throws.

Comment (iamjackslayer) [reply src/main/java/seedu/address/model/Model.java]: noted with thanks

**Comment by iamjackslayer** [comment]
Work in progress.

**Pull request closed without merging**